### ISSUE TITLE
Invalid validation of related item (M2A)

### ISSUE THREAD

**Issue by robinbisping** [directus/directus#12603]
### Describe the Bug

When linking an existing item in an M2A field and editing the linked item in the drawer, the fields disappear on saving if the values have not been modified.

This is annoying because fields added to a junction table cannot be edited.

This issue seems very similar to the bug in the M2M field: #12599 

### To Reproduce

1. Create two collections:
  ![Screenshot from 2022-04-06 14-22-03](https://user-images.githubusercontent.com/4021583/161973798-6bd79393-378a-480b-89b8-3678320df16a.png)

  - Collection Home contains an M2A field, linking the collection Block
    ![Screenshot from 2022-04-06 14-22-09](https://user-images.githubusercontent.com/4021583/161973959-e65ce29b-7a11-4d5d-a76c-a892c9433630.png)

  - Collection Block contains one field (name) that is required
    ![Screenshot from 2022-04-06 14-22-15](https://user-images.githubusercontent.com/4021583/161974074-8f8bc98e-c150-48da-8f15-05b319fa8a2c.png)

3. Create a Block item and save it
4. Create a Home item and add the previously created Block item to the M2A field (do not save)
5. Open the block item in a drawer by clicking on it in the M2A field

  ![Screenshot from 2022-04-06 14-26-45](https://user-images.githubusercontent.com/4021583/161974417-b0cdd17a-f6af-4ac4-9a7d-de99ace3157a.png)

6. Without editing the value, click save in the drawer
7. The fields disappear, but the drawer does not close
  ![Screenshot from 2022-04-06 14-21-53](https://user-images.githubusercontent.com/4021583/161974544-34e09d41-5b78-41c1-8862-6f48c35b78bc.png)
 

### Errors Shown

```
TypeError: $.field is undefined
    lM/</<["validation-error"]< http://0.0.0.0:8055/admin/assets/index.5bac3394.entry.js:43
    lM http://0.0.0.0:8055/admin/assets/index.5bac3394.entry.js:43
    KB http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    lM http://0.0.0.0:8055/admin/assets/index.5bac3394.entry.js:43
    v5 http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    pt http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    run http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    ut http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Oe http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    gr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Cr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Se http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    gr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Cr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    ee http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    $ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    gr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Cr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Se http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    gr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Cr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    ee http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    $ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    gr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Cr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    ee http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    $ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    gr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Cr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    ee http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    $ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    gr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Cr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Se http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    gr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Cr http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    ee http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    $ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    pt http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    run http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Ph http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    DE http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    promise callback*TE http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    Sb http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    ot http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    R7 http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    p5 http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    set value http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    te http://0.0.0.0:8055/admin/assets/index.5bac3394.entry.js:32
    Ph http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    W1 http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    nB http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/index.5bac3394.entry.js:43
    Ph http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    W1 http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    r http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    Ld http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    XH http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    QH http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    q http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    $ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    X http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    q http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    $ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    pt http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    run http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    mt http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    je http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Oe http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    X http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Se http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    X http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Se http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    X http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Se http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    X http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    q http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    $ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
vendor.ef99adf3.js:5:1162

TypeError: $.field is undefined
    lM/</<["validation-error"]< http://0.0.0.0:8055/admin/assets/index.5bac3394.entry.js:43
    lM http://0.0.0.0:8055/admin/assets/index.5bac3394.entry.js:43
    KB http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    lM http://0.0.0.0:8055/admin/assets/index.5bac3394.entry.js:43
    v5 http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    pt http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    run http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Ph http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    DE http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    promise callback*TE http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    Sb http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    ot http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    R7 http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    p5 http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    set value http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    te http://0.0.0.0:8055/admin/assets/index.5bac3394.entry.js:32
    Ph http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    W1 http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    nB http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/index.5bac3394.entry.js:43
    Ph http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    W1 http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    r http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    Ld http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    XH http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    QH http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:5
    q http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    $ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    X http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    q http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    $ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    pt http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    run http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    mt http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    je http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Oe http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    X http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Se http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    X http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Se http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    X http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    Se http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    X http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    q http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    $ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
    _ http://0.0.0.0:8055/admin/assets/vendor.ef99adf3.js:2
vendor.ef99adf3.js:5:1162

```

### What version of Directus are you using?

9.8.0

### What version of Node.js are you using?

16.13.1

### What database are you using?

Postgres 14

### What browser are you using?

Firefox

### How are you deploying Directus?

running locally

**Comment by rijkvanzanten** [comment]
I'm fairly sure this is caused by the same underlying issue, so lets track both these issues in #12599 🙂

**Comment by azrikahar** [comment]
I believe this turns out to be a different issue from #12599. It's not related to validation as I was able to reproduce this issue back in `v9.5.2`, whereas the PR #12001 was merged in `v9.7.0`.
